hedera-node/src/main/java/com/hedera/services/store/models/Token.java

```diff
@@ -92,7 +94,19 @@ public void mint(TokenRelationship treasuryRel, long amount) {
 		changeSupply(treasuryRel, +amount, INVALID_TOKEN_MINT_AMOUNT);
 	}
 
-	public OwnershipTracker mint(OwnershipTracker ownershipTracker, TokenRelationship treasuryRel, List<ByteString> metadata, final RichInstant creationTime) {
+	public void burn(final OwnershipTracker ownershipTracker, final TokenRelationship treasuryRelationship, final List<Long> serialNumbers){
+		validateTrue( type == TokenType.NON_FUNGIBLE_UNIQUE, INVALID_TRANSACTION_BODY, () ->
```

Comment:
If token type is not NFT we should throw with `FAIL_INVALID`. 
`FAIL_INVALID` is thrown when we should assert that something is true and it should be true unless we have bug in our code. This is the case here -> we won't get to this burn if the type of the token was not NFT (computed in transition logic)